src/components/UriViewer.js

```diff
@@ -183,6 +187,25 @@ const UriViewer = _.flow(
         })
         const metadata = await loadObject(googleProject, bucket, name)
         setMetadata(metadata)
+      } else if (isAzureUri(uri)) {
+        const provider = AzureBlobStorageFileBrowserProvider(
+          { workspaceId: workspace.workspace.workspaceId, pageSize: 1 }
+        )
+        const getFile = await provider.getFilesInDirectory('').then(files => {
+          return files.items[0]
+        })
+        await provider.getDownloadUrlForFile(getFile.path).then(url => {
```

Comment:
Style nit: Instead of handling the result in a `then` callback, this could be written as
```js
const url = await provider.getDownloadUrlForFile(...)
const metadata = { ... }
...
```

Reply:
This line is where using a file browser provider is problematic. `getDownloadUrlForFile` takes a _path_, not a _URL_. This is because file browser providers can assume that any files they will have to deal with are in the workspace bucket / storage container. That assumption isn't true for data tables.

For files in another storage container, this would return an incorrect URL (because file browser providers construct the download URL using the workspace bucket/storage container). Even worse, if file is in another storage container and the workspace storage container has a file at the same path, the download link would appear to work but it would download a different file than expected.